nautobot/project-static/js/forms.js

```diff
@@ -632,18 +632,19 @@ function initializeDynamicFilterForm(context){
 
     // Remove applied filters
     this_context.find(".remove-filter-param").on("click", function(){
-        let query_params = location.search;
+        let query_params = new URLSearchParams(location.search);
         let type = $(this).attr("data-field-type");
         let field_value = $(this).attr("data-field-value");
-        let query_string = location.search.substr(1).split("&");
 
         if (type === "parent") {
-            query_string = query_string.filter(item => item.search(field_value) < 0);
+            // Remove all instances of this query param
+            query_params.delete(field_value);
         } else {
+            // Remove this specific instance of this query param
             let parent = $(this).attr("data-field-parent");
-            query_string = query_string.filter(item => item.search(parent + "=" + field_value) < 0)
```

Comment:
Before the security fix above, with a query of `?name=Hello World`, we'd incorrectly have `data-field-value="Hello" world=""` on the `remove-filter-param` element. This would result in `field_value = "Hello"` and `query_string = ["name=Hello%20World"]`, but since we were (incorrectly IMHO) doing a fuzzy match here (`item.search("name=Hello")`) to find the query param to remove, it still "worked".

After the security fix, with corrected `data-field-value="Hello World"`, we'd correctly get `field_value = "Hello World"` but this then wouldn't match the URL-encoded `query_string` array, which had the net result that clicking the "x" button in the UI wouldn't actually remove such a query parameter from the view. (Yay! Integration tests actually caught a bug!)

The fix here is to not use string-splitting and regex-searching to manipulate the query_params, but instead to use the dedicated `URLSearchParams` interface available in all modern browsers.

Reply:
Did you test these additional fixes on chrome and firefox?

Reply:
Just Firefox so far, but will try on Chrome as well. According to https://developer.mozilla.org/en-US/docs/Web/API/URLSearchParams#browser_compatibility the only one we should need to worry about is Samsung's mobile browser (??)

Reply:
Anyone using that browser also has chrome available so it's probably ok?

Reply:
Actually, I'll test on samsung internet

Reply:
Works fine in Chrome too.